addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -319,6 +319,50 @@ def test_partner_assigned_to_next_order_coupon(self):
             "The partner should be set on the coupon with program type 'next_order_coupons'"
         )
 
+    def test_public_partner_updated_in_next_order_coupon(self):
+        """ Test the update of a partner on coupons with program type `next_order_coupons`.
+
+        1. Create a loyalty program of type `next_order_coupons`.
+        2. Create a sale order for a public user and add a product to it.
+        3. Apply the loyalty program to the sale order.
+        4. Verify that the generated coupon is assigned to the public user.
+        5. Change the partner.
+        6. Verify that the generated coupon was updated to this new user.
+        """
+        loyalty_program = self.env['loyalty.program'].create({
+            'name': '10% Discount on Next Order',
+            'program_type': 'next_order_coupons',
+            'applies_on': 'future',
+            'trigger': 'auto',
+            'rule_ids': [Command.create({})],
+            'reward_ids': [Command.create({
+                'reward_type': 'discount',
+                'discount': 10,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })],
+        })
+        order = self.empty_order.copy()
+        order.write({
+            'partner_id': self.env.ref('base.public_user').id,
```

Comment:
```suggestion
            'partner_id': self.env.ref('base.public_partner').id,
```
This one's on me, but the only reason this worked is because `public_user` & `public_partner` both happen to have the same `id` value.